ui/src/pages/tasks/TaskDraw.tsx

```diff
@@ -5,40 +5,62 @@ import TaskCard from '@components/TaskCard';
 import { Navigate, useNavigate, useParams } from 'react-router-dom';
 import LoadingScreen from '@components/LoadingScreen';
 import paths from '@routes/paths';
-import { useGetTaskQuery } from 'api/tasks/hooks';
+import { useClassifyImageMutation, useGetTaskQuery } from 'api/tasks/hooks';
+import { ReactSketchCanvasRef } from 'react-sketch-canvas';
+import React from 'react';
+import { convertImageToBlob } from '@lib/downloadImage';
 
 export default function TaskDraw() {
-  const { id } = useParams();
   const startTime = new Date().getTime();
+  const { id } = useParams();
   const navigate = useNavigate();
+  const ref = React.createRef<ReactSketchCanvasRef>();
+
+  const { data: taskData, isLoading: taskLoading, isError: taskError } = useGetTaskQuery(Number(id));
+  const { mutateAsync: resultMutate, isPending: resultLoading, isError: resultError } = useClassifyImageMutation();
+
+  if (taskLoading || resultLoading) return <LoadingScreen />;
+  if (taskError || resultError) return <div className="error">{t('firebase.unknown-error')}</div>;
+  if (!taskData) return <Navigate to={paths.ROOT} />;
+
+  const getImageBlob = async () => {
+    const image = ref.current?.exportImage;
+    if (image) return convertImageToBlob(await image('png'));
+    return null;
+  };
+
+  const handleSubmit = async () => {
+    const elapsedTime = (new Date().getTime() - startTime) / 1000;
+    const image = await getImageBlob();
 
-  const { data, isLoading, isError } = useGetTaskQuery(Number(id));
+    if (!image || !id) return;
 
-  if (isLoading) return <LoadingScreen />;
-  if (isError) return <div className="error">{t('firebase.unknown-error')}</div>;
-  if (!data) return <Navigate to={paths.ROOT} />;
+    const resultData = await resultMutate({
```

Comment:
zakładamy że to się zawsze powiedzie? Nie musimy dodać żadnego try catch?

Reply:
Jeśli wystąpi błąd, to wtedy przechwyci wtedy w resultError będzie błąd i można to wyżej obsłużyć.

Reply:
niekoniecznie, tutaj może wystąpić error podczas parsowania jsona który może wykrzaczyć aplikację